### ISSUE TITLE
Unable to use fit_onecycle on a model I compiled due to optimizer

### ISSUE THREAD

**Issue by AdamBrownLab** [amaiya/ktrain#42]
I'm trying experiment with the onecycle policy of the (AMAZING) ktrain library.
I took the old cats vs dogs via augmentation from keras [blog ](https://blog.keras.io/building-powerful-image-classification-models-using-very-little-data.html)
https://gist.github.com/fchollet/0830affa1f7f19fd47b06d4cf89ed44d, and I want to see how the training policy changes the results.

However, the learner.fit_onecycle return the error `'RMSprop' object has no attribute 'beta_1'`.
How should I compile my model, to allow onecycle to do its magic? 
Is there a way to allow it to override the preexisting optimizer compiled into the model?

It is not possible to compile without an optimizer or to use the ktrain fit_onecycle without first compiling the model.

Many thanks :)

**Comment by AdamBrownLab** [comment]
Oh, is this because I did not compile the model with the adam optimizer? 
It does run when I try it - but are the advantages of the policy are still in effect?

**Comment by amaiya** [comment]
By default, `fit_onecycle` (and `autofit`) will cycle the momentum along with learning rate for any Adam-like optimizer with `beta_1` parameter.  You can use `rmsprop` if you set `cycle_momentum=False` in the call to `learner.fit_onecycle`.  When using RMSProp with `cycle_momentum=False`, the learning rate will still follow a 1cycle policy, but the momentum won't be cycled.

However, the docstring for `fit_oneycyle` needs to be corrected and there should probably be a check and perhaps graceful warning instead of a hard error if someone uses an optimizer other than Adam with `cycle_momentum=True`.  So, I'm marking this as a Bug to be fixed in next release.

**Comment by AdamBrownLab** [comment]
Glad to be of service, and thank you.

**Comment by amaiya** [comment]
Fixed in 0.8.0 - thanks
